testsuite/tests/index/git-local/test.py

```diff
@@ -0,0 +1,220 @@
+        path = path.removeprefix("/private")
+    assert_match(
+        rf".*\d+.*{re.escape(name)}.*{re.escape(url)}.*{re.escape(path)}",
+        run_alr("index", "--list").out
+    )
+
+def check_index_works():
+    """
+    Use the currently configured index to get, build and run the 'hello' crate,
+    asserting success.
+    """
+    run_alr("get", "hello")
+    deploy_dir = run_alr("get", "hello", "--dirname").out.strip()
+    os.chdir(deploy_dir)
+    assert_eq("Hello, world!\n", run_alr("run").out)
+    os.chdir("..")
+    shutil.rmtree(deploy_dir)
+
+
+# Add all necessary paths to the index. We do this now because they need to be
+# absolute paths; relative paths don't work for git repo indexes. Note that TOML
+# files require backslashes (in windows paths) to be escaped.
+my_crates_path = os.path.join(TEST_ROOT_DIR, "my_crates")
+hello_manifest_path = os.path.join(
+    "my_index", "index", "he", "hello", "hello-1.0.1.toml"
+)
+libhello_manifest_path = os.path.join(
+    "my_index", "index", "li", "libhello", "libhello-1.0.0.toml"
+)
+hello_path = os.path.join(my_crates_path, "hello").replace("\\", "\\\\")
+libhello_path = os.path.join(my_crates_path, "libhello").replace("\\", "\\\\")
+with open(hello_manifest_path, "a") as f:
+    f.write(f'url = "file:{hello_path}"\n')
+with open(libhello_manifest_path, "a") as f:
+    f.write(f'url = "file:{libhello_path}"\n')
+
+
+# Test adding my_index as a simple directory index.
+run_alr("index", "--name", "my_index", "--add", "my_index")
+check_index_is_configured("my_index", f"file:{MY_INDEX_PATH}", MY_INDEX_PATH)
+# Verify this hasn't created an unnecessary copy in the alr-config directory
+# (only an index.toml file which redirects to the existing copy).
+index_copy_path = os.path.join("alr-config", "indexes", "my_index")
+assert_eq(["index.toml"], os.listdir(index_copy_path))
+# Check that the index can be used to get and build a crate.
+check_index_works()
+# Clean up for next test case.
+run_alr("index", "--del", "my_index")
+
+# Verify same result with --arg=value form.
+run_alr("index", "--name=my_index", "--add=my_index")
+check_index_is_configured("my_index", f"file:{MY_INDEX_PATH}", MY_INDEX_PATH)
+assert_eq(["index.toml"], os.listdir(index_copy_path))
+check_index_works()
+run_alr("index", "--del", "my_index")
+
+# Verify the same result with a `file:` URL equivalent.
+run_alr("index", "--name", "my_index", "--add", "file:my_index")
+check_index_is_configured("my_index", f"file:{MY_INDEX_PATH}", MY_INDEX_PATH)
+assert_eq(["index.toml"], os.listdir(index_copy_path))
+check_index_works()
+run_alr("index", "--del", "my_index")
+
+# Verify that a `git+file:` URL fails.
+p = run_alr(
+    "index", "--name", "my_index", "--add", "git+file:my_index",
+    complain_on_error=False
+)
+unix_pattern = "repository '.*my_index' does not exist"
+windows_pattern = "'.*my_index' does not appear to be a git repository"
+assert_match(f".*({unix_pattern}|{windows_pattern})", p.out)
```

Comment:
Is this dependent on the git version? I'm afraid this might break in the future without warning.

Reply:
Fair enough. It now only checks the error Alire produces when the `git` command fails.